### PR TITLE
Basic stats

### PR DESCRIPTION
**Pull request by adkauth**
There's a stats component to hold player stats. Game jam it up and modify it from wherever you feel like.
Also you can click on squares during prep to add towers.

### CODE REVIEW

File: src/components.js [hunk 1]
```diff
@@ -26,19 +26,21 @@ export class Tower extends Component {};
 Tower.schema = {
   x: { type: Types.Number },
   y: { type: Types.Number },
-  damage: { type: Types.Number },
-  range: { type: Types.Number },
-  rateOfFire: { type: Types.Number },
+  range: { type: Types.Number, default: 1 },
+  damage: { type: Types.Number, default: 10 },
+  rateOfFire: { type: Types.Number, default: 3 },
```
Comment (DingoEatingFuzz) [inline src/components.js:31]: defaults, nice.

File: src/components.js [hunk 2]
```diff
@@ -26,19 +26,21 @@ export class Tower extends Component {};
...
+  cooldown: { type: Types.Number },
   areaOfEffect: { type: Types.Boolean },
-  _pool: {type: Types.Boolean },
+  price: {type: Types.Number, default: 50},
+  _pool: {type: Types.Boolean }, //gamejam btw
```
Comment (DingoEatingFuzz) [inline src/components.js:35]: smh

File: src/components.js [hunk 3]
```diff
@@ -55,3 +57,11 @@ EnemyEmitter.schema = {
...
   cooldown: { type: Types.Number },
 }
+
+export class Stats extends Component {};
+Stats.schema = {
+  points: { type: Types.Number },
+  health: { type: Types.Number },
+  money: { type: Types.Number },
+  BAWN: { type: Types.Boolean },
+}
```
Comment (DingoEatingFuzz) [inline src/components.js:67]: BAWN????

**Comment by adkauth** [comment]
Oh and now turrets shoot enemies too

File: src/systems/enemies.js [hunk 4]
```diff
@@ -13,14 +13,20 @@ export default class EnemiesSystem extends System {
...
 
   updateEnemies() {
+    let stats = this.queries.stats.results[0].getMutableComponent(Stats);
     this.queries.enemies.results.forEach(ent => {
       const location = ent.getMutableComponent(Location);
       const path = ent.getComponent(Path);
       const enemy = ent.getComponent(Enemy);
+      if (enemy.health <= 0) {
+          console.log(`${enemy.name} has died`);
+          ent.removeAllComponents();
```
Comment (DingoEatingFuzz) [inline src/systems/enemies.js:23]: Money goes here?

File: src/systems/screenRenderer.js [hunk 5]
```diff
@@ -10,16 +10,17 @@ export default class GridRenderSystem extends System {
...
     switch (activeScene.scene.key){
       case "menu":
-        activeScene.add.text(700,700,"menu")
+        //activeScene.add.text(700,700,"menu")
         break;
       case "prep":
-        activeScene.add.text(700,700,"prep")
-          break;
+        //activeScene.add.text(700,700,"prep")
+        //  break;
```
Comment (DingoEatingFuzz) [inline src/systems/screenRenderer.js:20]: GAAAAAAAAAAAAAAAAME JAM

File: src/systems/towers.js [hunk 6]
```diff
@@ -24,9 +24,36 @@ export default class TowerSystem extends System {
...
+
+  //once working we deal with cooldown/ROF
+  makeTheTowersDoTheShootyThing() {
+    this.queries.legitTowers.results.forEach(tEnt => {
+        const tower = tEnt.getMutableComponent(Tower);
+        if (tower.cooldown === 0) {
+            this.queries.enemies.results.forEach(eEnt => {
+                const enemyLoc = eEnt.getComponent(Location);
+                //check cooldown again or do something "fancy"
+                if (tower.cooldown === 0 && Math.hypot(tower.x - enemyLoc.x, tower.y - enemyLoc.y) <= tower.range) {
```
Comment (DingoEatingFuzz) [inline src/systems/towers.js:36]: woah, `Math.hypot` is new

File: src/systems/towers.js [hunk 7]
```diff
@@ -24,9 +24,36 @@ export default class TowerSystem extends System {
...
+                    eEnt.getMutableComponent(Enemy).health -= tower.damage;
+                    tower.cooldown = tower.rateOfFire;
```
Comment (DingoEatingFuzz) [inline src/systems/towers.js:38]: It would be nice to exit the loop early, but game jam.

File: src/world.js [hunk 8]
```diff
@@ -30,6 +30,8 @@ export default class LudumWorld {
 
     // This entity should get a path that goes all the way down and back up to avoid the wall
     this.world.createEntity().addComponent(components.Path, { from: [0,0], to: [9,0] });
+
+    this.world.createEntity().addComponent(components.Stats, {points: 0, health: 69, money: 420, BAWN: false});
```
Comment (DingoEatingFuzz) [inline src/world.js:34]: Not BAWN 😭

**Review by DingoEatingFuzz — APPROVED** [review]
Good enough!

**Pull request merged**